src/toc.tsx

```diff
@@ -55,45 +90,34 @@ export class TableOfContents extends Widget {
     if (this.generator && this.generator.toolbarGenerator) {
       this._toolbar = this.generator.toolbarGenerator();
     }
-
-    // Dispose an old activity monitor if it existsd
+    // Dispose an old activity monitor if one existed...
     if (this._monitor) {
       this._monitor.dispose();
       this._monitor = null;
     }
-    // If we are wiping the ToC, update and return.
+    // If we are wiping the ToC, update and return...
     if (!this._current) {
-      this.updateTOC();
+      this.update();
       return;
     }
-
-    // Find the document model associated with the widget.
+    // Find the document model associated with the widget:
     const context = this._docmanager.contextForWidget(this._current.widget);
     if (!context || !context.model) {
       throw Error('Could not find a context for the Table of Contents');
     }
-
-    // Throttle the rendering rate of the table of contents.
+    // Throttle the rendering rate of the table of contents:
     this._monitor = new ActivityMonitor({
       signal: context.model.contentChanged,
       timeout: RENDER_TIMEOUT
     });
     this._monitor.activityStopped.connect(this.update, this);
-    this.updateTOC();
+    this.update();
   }
 
   /**
-   * Handle an update request.
+   * Updates a table of contents.
    */
-  protected onUpdateRequest(msg: Message): void {
-    // Don't bother if the TOC is not visible
-    /* if (!this.isVisible) {
-      return;
-    } */
-    this.updateTOC();
-  }
-
-  updateTOC() {
+  update() {
```

Comment:
This overrides an existing `update()` method, which we don't want to hide. Let's revert the name change, or put this logic into the `onUpdateRequest` function.

Reply:
Moved logic to the `onUpdateRequest` function. Thanks! Did not realize I clobbered an existing method.